techtv2ovs/utils.py

```diff
@@ -0,0 +1,314 @@
+    https://github.com/TechTV/TechTV/blob/master/app/models/collection/moira_controllable.rb#L76
+
+    Args:
+        cid (int): The ID of a TechTVCollection object
+        name (str): The name of the TechTVCollection object
+
+    Returns:
+        str: The name of the moira list for the collection
+    """
+    # Replace certain special characters with dashes
+    return 'techtv-{}-{}'.format(
+        cid,
+        re.sub(r'\-+', '-', (re.sub(r'[^A-Za-z0-9\-_]+', '', re.sub(r'[\s\'\.,\-\/\\@]+', '-', name))))[0:32].lower()
+    )
+
+
+def get_s3_videos(s3_prefix):
+    """
+    Get all the available S3 videofile objects for a TechTVVideo
+
+    Args:
+        s3_prefix (str): The TechTVVideo external id to use as the S3 bucket prefix to filter by
+
+    Returns:
+        list: A list of S3 objects
+    """
+    s3client = boto3.client('s3')
+    meta = s3client.list_objects_v2(Bucket=TTV_VIDEO_BUCKET, Prefix=s3_prefix)
+    return meta['Contents'] if 'Contents' in meta else []
+
+
+def get_owner(email):
+    """
+    Get/create a user for emails ending with `@mit.edu`, otherwise return a default user
+
+    Args:
+        email (str): An email address
+
+    Returns:
+        User: A User object.
+    """
+    if email and email.endswith('@mit.edu'):
+        user, _ = User.objects.get_or_create(username=email, defaults={
+            'email': email
+        })
+        return user
+    return User.objects.get(username=settings.LECTURE_CAPTURE_USER)
+
+
+def remove_tags(txt):
+    """
+    Replace any HTML tags with spaces, HTML-encoded characters with unicode
+
+    Args:
+        txt (str): Text to remove HTML tags from.
+
+    Returns:
+        str: String with HTML tags/characters removed.
+    """
+    if not txt:
+        return ''
+    return normalize("NFKD", unescape(strip_tags(txt)))
+
+
+class TechTVImporter:
+    """
+    Class for handling TechTV to OVS imports
+    """
+
+    def __init__(self, db_user=None, db_pw=None, db_name='techtv', db_host='127.0.0.1',
+                 collection=None, aws=False, protected=1, output=sys.stdout):
+        """
+        Assign class properties based on kwargs
+
+        Args:
+            db_user (str): MySQL user
+            db_pw (str): MySQL password
+            db_name (str): MySQL database
+            db_host (str): MySQL host
+            collection (int): TechTV collection id
+            aws (bool): Copy S3 files from TechTV to OVS
+            protected (int): Process protected (1) or unprotected (0) collections
+            output (object): Where to direct output to
+        """
+        self.db_user = db_user
+        self.db_pw = db_pw
+        self.db_name = db_name
+        self.db_host = db_host
+        self.collection = collection
+        self.aws = aws
+        self.protected = protected
+        self.output = output
+
+    def run(self):
+        """
+        Run the import
+        """
+        with MySQLdb.connect(
```

Comment:
Can you write a test for this function? I'm mostly concerned about the last line, that there's an assertion that `self.etl_graph()` is the thing which gets executed

Reply:
✅